templates/mongodb.conf.2.6.erb

```diff
@@ -150,11 +150,17 @@ operationProfiling.mode: all
 operationProfiling.slowOpThresholdMs: <%= @slowms %>
 <% end -%>
 
-
-<% if @set_parameter -%>
+<% if @set_parameter and @set_parameter_array.nil? -%>
 setParameter: <%= @set_parameter %>
 <% end -%>
 
+<% if @set_parameter_array -%>
+setParameter:
+<% @set_parameter_array.each do |v| -%>
```

Comment:
I think we can do something like this and afterwards drop both if statements in the erb template and the logic in the init.pp:
```suggestion
<% [@set_parameter_array].flatten.each do |v| -%>
```

Reply:
Thank you for the hint, changed